modules/configuring-default-route-timeouts-globally.adoc

```diff
@@ -0,0 +1,111 @@
+//
+// * knative-serving/config-applications/configuring-revision-timeouts.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="configuring-default-route-timeouts-globally_{context}"]
+= Configuring the default route timeouts globally
+
+By configuring the route timeouts globally, you can ensure consistent timeout settings across all services, simplifying management for workloads that have similar timeout needs and reducing the need for individual adjustments.
+
+You can configure the route timeouts globally by updating the `ROUTE_HAPROXY_TIMEOUT` environment value in your `serverless-operator` subscription and updating the `max-revision-timeout-seconds` field in your `KnativeServing` custom resource (CR). This applies the timeout changes across all Knative services, and you can deploy services with specific timeouts up to the maximum value set.
+
+.Procedure
+
+. Set the value of `ROUTE_HAPROXY_TIMEOUT` in your subscription to your required timeout in seconds by running the following command:
++
+.Setting the `ROUTE_HAPROXY_TIMEOUT` value to 900 seconds
+[source,terminal]
+----
+$ oc patch subscription.operators.coreos.com serverless-operator -n openshift-serverless --type='merge' -p '{"spec": {"config": {"env": [{"name": "ROUTE_HAPROXY_TIMEOUT", "value": "900"}]}}}'
+----
++
+The `ROUTE_HAPROXY_TIMEOUT`  environment variable is managed by the Serverless Operator and by default is set to `600`. Set the value of `ROUTE_HAPROXY_TIMEOUT` in your subscription to your required timeout in seconds by running the following command. Note that this causes pods to be redeployed in the `openshift-serverless` namespace.
++
+[NOTE]
+====
+If you created your routes manually and disabled auto-generation with the `serving.knative.openshift.io/disableRoute` annotation, you can configure the timeouts directly in the route definitions.
+====
+
+. Set the maximum revision timeout in your `KnativeServing` CR:
++
+.`KnativeServing` CR with `max-revision-timeout-seconds` set to 900 seconds
+[source,yaml]
+----
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+spec:
+  config:
+    defaults:
+      max-revision-timeout-seconds: "900"
+#...
+----
++
+The Serverless Operator automatically adjusts the `terminationGracePeriod` value of the activator to the set maximum revision timeout value to avoid request termination in cases where activator pods are being terminated themselves.
++
+. Optional: Verify that the timeout has been set by running the following command: 
++
+[source,terminal]
+----
+$ oc get deployment activator -n knative-serving -o jsonpath="{.spec.template.spec.terminationGracePeriodSeconds}"
+----
+
+. If necessary for your cloud provider, adjust the load balancer timeout by running the following command:
++
+.Load balancer timeout adjustment for AWS Classic LB 
+[source,terminal]
+----
+$ oc -n openshift-ingress-operator patch ingresscontroller/default --type=merge --patch='
```

Comment:
Hi, these need to be broken apart with a `\` as described in the [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#backslash-for-long-lines-of-code). That ensures that they display syntax highlighting properly and are copied correctly if someone uses the "click to copy" function. For example:

```
$ oc -n openshift-ingress-operator patch ingresscontroller/default --type=merge --patch= \
  '{"spec":{"endpointPublishingStrategy":  \
  {"scope":"External", "providerParameters":{"type":"AWS", "aws":  \
  {"type":"Classic", "classicLoadBalancer":  \
  {"connectionIdleTimeout":"20m"}}}}}}}'
```